test_mcp_ecosystem_expansion.py

```
@@ -55,434 +58,475 @@ async def run_all_tests(self) -> Dict[str, Any]:
+    elif summary["success_rate"] >= 0.7:
         print("⚠️  Good progress! Most components are working.")
         print("   Review failed tests and address any critical issues.")
     else:
         print("❌ Several issues detected. Review and fix failed tests.")
         print("   Focus on core functionality before production deployment.")
-    
+
     # Save detailed report
     report_file = f"mcp_ecosystem_test_report_{int(time.time())}.json"
-    with open(report_file, 'w') as f:
+    with open(report_file, "w") as f:
```

Ruff (0.11.9)

Async functions should not open files with blocking methods like `open`

(ASYNC230)